components/constellation/constellation.rs

```diff
@@ -2624,11 +2625,21 @@ where
     }
 
     fn handle_key_msg(&mut self, ch: Option<char>, key: Key, state: KeyState, mods: KeyModifiers) {
-        // Send to the explicitly focused pipeline. If it doesn't exist, fall back to sending to
-        // the compositor.
-        match self.focus_pipeline_id {
-            Some(pipeline_id) => {
+        // Send to the focused browsing contexts' current pipeline.  If it
+        // doesn't exist, fall back to sending to the compositor.
+        match self.focused_browsing_context_id {
+            Some(browsing_context_id) => {
                 let event = CompositorEvent::KeyEvent(ch, key, state, mods);
+                let pipeline_id = {
```

Comment:
This can be:
```rust
let pipeline_id = match self.browsing_contexts.get(&browsing_context_id) {
    Some(browsing_context) => browsing_context.pipeline_id,
    None => return warn!(...);
};
```